DatWeatherDoe/UI/Decorator/Text/RiseSet/RiseSetTextBuilder.swift

```diff
@@ -0,0 +1,68 @@
+//
+//  RiseSetTextBuilder.swift
+//  DatWeatherDoe
+//
+//  Created by Markus Markus on 7/26/22.
+//  Copyright © 2022 Markus Mayer.
+//
+
+import Foundation
+
+final class RiseSetTextBuilder {
+
+    private let initial: String
+    private let sunset: TimeInterval
+    private let sunrise: TimeInterval
+
+    private let upArrowStr = "⬆"
+    private let downArrowStr = "⬇"
+
+    init(
+        initial: String = "",
+        sunset: TimeInterval = 0,
+        sunrise: TimeInterval = 0
+    ) {
+        self.initial = initial
+        self.sunset = sunset
+        self.sunrise = sunrise
+    }
+
+    func build() -> String {
+        let riseSetString = buildRiseSet()
+        if riseSetString == "" {
+            return initial
+        }
+        return (initial != "") ? "\(initial) \(riseSetString)" : riseSetString
+    }
+
+    private func buildRiseSet() -> String {
+        var ret: String = ""
+        let sunRiseText = buildFormattedString(ts: sunrise)
+        let sunSetText = buildFormattedString(ts: sunset)
+
+        if sunRiseText != "" {
+            ret = "\(upArrowStr)\(sunRiseText)"
+        }
+        if sunSetText != "" {
+            if ret != "" {
+                ret += " "
+            }
+            ret += "\(downArrowStr)\(sunSetText)"
+        }
+
+        return ret
+    }
+
+    private func buildFormattedString(ts: TimeInterval) -> String {
+        if ts == 0 {
```

Comment:
Will this case ever be possible if we remove the defaults from the initializer? As long as the API guarantees values for sunset and sunrise, we can eliminate this case

Reply:
Not yet. However, I have some other changes that build on top of what I submitted. And with those changes, it will be possible. This "if" statement can be removed for now and could be bundled with the additional changes (which is displaying sunrise and/or sunset in the status bar).